src/engine/SCons/Script/Main.py

```diff
@@ -1099,17 +1100,36 @@ def _main(parser):
                                           targets, target_top)
 
     else:
-        # Write the build graph
-        graph = os.environ.get("SCONS_GRAPH")
-        if graph:
+        scons_mode = os.environ.get("SCONS_MODE")
+        if options.mode == "vulcan" or scons_mode == "vulcan":
+            # Write the build graph
+            graph = "scons.gv"
             SCons.Spirent.GraphWriter().write(graph, fs.Top)
-
-        # Build the targets
-        nodes = _build_targets(fs, options, targets, target_top)
-        if not nodes:
-            revert_io()
-            print('Found nothing to build')
-            exit_status = 2
+            # Get the vulcan command line options
+            if options.mode == "vulcan":
```

Comment:
At this point in the code, we know the mode is `vulcan` and there is no need to check it again. Options should come from the environment or command line in that order of precedence.